static/app/views/dashboardsV2/widgetCard/genericWidgetQueries.tsx

```diff
@@ -0,0 +1,325 @@
+        query.columns = query.columns.filter(field => !!field);
+        return query;
+      })
+      .reduce(
+        ([names, queries]: [string[], Omit<WidgetQuery, 'name'>[]], {name, ...rest}) => {
+          names.push(name);
+          queries.push(rest);
+          return [names, queries];
+        },
+        [[], []]
+      );
+
+    const [widgetQueryNames, widgetQueries] = widget.queries
+      .map((query: WidgetQuery) => {
+        query.aggregates = query.aggregates.filter(
+          field => !!field && field !== 'equation|'
+        );
+        query.columns = query.columns.filter(field => !!field && field !== 'equation|');
+        return query;
+      })
+      .reduce(
+        ([names, queries]: [string[], Omit<WidgetQuery, 'name'>[]], {name, ...rest}) => {
+          names.push(name);
+          queries.push(rest);
+          return [names, queries];
+        },
+        [[], []]
+      );
+
+    if (
+      widget.limit !== prevProps.widget.limit ||
+      !isEqual(widget.displayType, prevProps.widget.displayType) ||
+      !isEqual(widget.interval, prevProps.widget.interval) ||
+      !isEqual(widgetQueries, prevWidgetQueries) ||
+      !isSelectionEqual(selection, prevProps.selection) ||
+      cursor !== prevProps.cursor
+    ) {
+      this.fetchData();
+      return;
+    }
+
+    if (
+      !this.state.loading &&
+      !isEqual(prevWidgetQueryNames, widgetQueryNames) &&
+      this.state.rawResults?.length === widget.queries.length
+    ) {
+      // If the query names has changed, then update timeseries labels
+
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState(prevState => {
+        const timeseriesResults = widget.queries.reduce((acc: Series[], query, index) => {
+          return acc.concat(
+            config.transformSeries!(prevState.rawResults![index], query, organization)
+          );
+        }, []);
+
+        return {...prevState, timeseriesResults};
+      });
+    }
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  private _isMounted: boolean = false;
+
+  async fetchTableData(queryFetchID: symbol) {
+    const {
+      widget,
+      limit,
+      config,
+      api,
+      organization,
+      selection,
+      cursor,
+      processRawTableResult,
+      onDataFetched,
+    } = this.props;
+    const responses = await Promise.all<[TableResponse, string, ResponseMeta]>(
+      widget.queries.map(query => {
+        let requestLimit: number | undefined = limit ?? DEFAULT_TABLE_LIMIT;
+        let requestCreator = config.getTableRequest;
+        if (widget.displayType === DisplayType.WORLD_MAP) {
+          requestLimit = undefined;
+          requestCreator = config.getWorldMapRequest;
+        }
+        return requestCreator!(
+          api,
+          query,
+          organization,
+          selection,
+          requestLimit,
+          cursor,
+          getReferrer(widget.displayType)
+        );
+      })
+    );
```

Comment:
should we try / throw errors here if the `config.getTableRequest` or `config.getWorldMapRequest` is expected but is undefined?

Reply:
I added this:
```js
        if (!requestCreator) {
          throw new Error(
            t('This display type is not supported by the selected dataset.')
          );
        }
```

and the try/catch of the parent `fetchData` call pulls out the error message